pkg/util/process_file_stats_others.go

```diff
@@ -0,0 +1,15 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build !linux
+// +build !linux
+
+package util
+
+import "github.com/go-openapi/errors"
+
+func GetProcessFileStats() (*ProcessFileStats, error) {
+	return nil, errors.NotImplemented("This function is only implemented for Linux-based OSes.")
```

Comment:
Lets use the error type `util.ErrNotImplemented` that already [exists](https://github.com/DataDog/datadog-agent/blob/0ab9e1a76b5c2ea840c15f1b3ed9fba029782dbf/pkg/process/util/util.go#L23).

Using a known error type like this would be helpful if we were passing into some 3rd party code that compares against it, but as-is, I don't see any value in using an error type from a 3rd party package.

In general, we should have a good reason before pulling in a new dependency, even though this dep exists today, its an indirect dependency (which means we depend on a library that depends on `go-openapi`) and using it directly like this would make it a direct dependency.